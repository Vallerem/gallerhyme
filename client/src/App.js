import React, { Component } from "react";
import "./App.css";
import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";

// import _ from "lodash";
import UsersList from "./views/user/UsersList";

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Router>
        <Switch>
          <Route
            exact
            path="/"
            component={() => (
              <h1>
                Home
                <Link to="/users"> To users</Link>
              </h1>
            )}
          />
          <Route exact path="/users" component={UsersList} />
          <Route exact path="/users/:id" component={() => <h1>Detal!!</h1>} />
          <Route path="/login" render={() => <h1> Login</h1>} />
          <Route path="/signup" render={() => <h1> Signup</h1>} />
          <Route render={() => <h1> Not found</h1>} />
        </Switch>
      </Router>
    );
  }
}

export default App;
