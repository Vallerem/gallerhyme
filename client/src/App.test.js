import App from "./App";
import { Route } from "react-router";

describe("<App />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("renders itself", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders all routes", () => {
    // Extremely redundantÂ ðŸ™ˆ
    const routesLength = wrapper.find(Route).length;
    expect(wrapper.find(Route)).toHaveLength(routesLength);
  });
});
