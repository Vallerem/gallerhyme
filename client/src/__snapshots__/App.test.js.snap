// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders itself 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/users"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/users/:id"
    />
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      path="/signup"
      render={[Function]}
    />
    <Route
      render={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
